src/main/java/at/hannibal2/skyhanni/features/garden/SensReducer.kt

```diff
@@ -0,0 +1,119 @@
+    private val gameSettings = mc.gameSettings
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!GardenAPI.inGarden()) {
+            if (isToggled && lastCheckCooldown.passedSince() > 1.seconds) {
+                lastCheckCooldown = SimpleTimeMark.now()
+                isToggled = false
+                restoreSensitivity()
+            }
+            return
+        }
+        if (isManualToggle) return
+        if (!config.enabled) {
+            if (isToggled) restoreSensitivity()
+            return
+        }
+        if (isToggled && config.inGround && !mc.thePlayer.onGround) {
+            restoreSensitivity()
+            isToggled = false
+            return
+        }
+        if (isToggled && config.inPlot && GardenAPI.onBarnPlot) {
+            restoreSensitivity()
+            isToggled = false
+            return
+        }
+        if (config.useKeybind) {
+            if (config.keybind.isKeyHeld() && !isToggled)  toggle(true)
+            else if (!config.keybind.isKeyHeld() && isToggled) toggle(false)
+        } else {
+            if (isHoldingTool() && !isToggled) toggle(true)
+            else if (!isHoldingTool() && isToggled) toggle(false)
+        }
+    }
+
+    @SubscribeEvent
+    fun onConfigInit(event: ConfigLoadEvent) {
+        config.reducingFactor.afterChange {
+            reloadSensitivity()
+        }
+    }
+
+    private fun reloadSensitivity() {
+        if (isToggled || isManualToggle) {
+            restoreSensitivity()
+            lowerSensitivity()
+        }
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!(isToggled || isManualToggle)) return
+        if (!config.showGUI) return
+        config.position.renderString("§eSensitivity Lowered", posLabel = "Sensitivity Lowered")
+    }
+
+    private fun isHoldingTool(): Boolean {
+        return GardenAPI.toolInHand != null
+    }
+
+    fun manualToggle() {
+        if (isToggled) {
+            LorenzUtils.chat("This command is disabled while holding a farming tool.")
+            return
+        }
+        isManualToggle = !isManualToggle
+        if (isManualToggle) {
+            lowerSensitivity(true)
+        } else restoreSensitivity(true)
+    }
+
+    private fun lowerSensitivity(showMessage: Boolean = false) {
+        storage.savedMouseloweredSensitivity = gameSettings.mouseSensitivity
+        val divisor = config.reducingFactor.get()
+        LorenzUtils.debug("dividing by $divisor")
+        val newSens =
+            ((storage.savedMouseloweredSensitivity + (1F / 3F)) / divisor) - (1F / 3F)
+        gameSettings?.mouseSensitivity = newSens
+        if (showMessage) LorenzUtils.chat("§bMouse sensitivity is now lowered. Type /shsensreduce to restore your sensitivity.")
+    }
+
+    private fun restoreSensitivity(showMessage: Boolean = false) {
+        gameSettings?.mouseSensitivity = SkyHanniMod.feature.storage.savedMouseloweredSensitivity
```

Comment:
I would prefer if you could *move* the sensitivity out of the config (Perhaps make savedMouseloweredSensitivty a `@Nullable` value). That way you can check first if you have a saved sensitivity at all. You could also use that saved sensitivity to then restore sensitivity after a crash or quit out.

Reply:
what if instead we do something like this
```kotlin
@SubscribeEvent
    fun onLogin(event: HypixelJoinEvent) {
        val divisor = config.reducingFactor.get()
        val expectedLoweredSensitivity = ((divisor * (gameSettings.mouseSensitivity + 1F / 3F)) - 1F / 3F)
        println("expected = $expectedLoweredSensitivity")
        if (storage.savedMouseloweredSensitivity == expectedLoweredSensitivity) {
            LorenzUtils.debug("Fixing incorrectly lowered sensitivity")
            restoreSensitivity()
        }
    }
```
maybe with a margin of error for float calculation